packages/zoe/test/swingsetTests/zoe/test-zoe.js

```diff
@@ -309,6 +309,66 @@ test('zoe - simpleExchange - valid inputs - no SES', async t => {
   }
 });
 
+const expectedSimpleExchangeUpdateLog = [
+  '=> alice, bob, carol and dave are set up',
+  'Order update: b:[], s:[]',
+  'The offer has been accepted. Once the contract has been completed, please check your payout',
+  'Order update: b:[], s:moola:3 for simoleans:4',
+  'The offer has been accepted. Once the contract has been completed, please check your payout',
+  'bobMoolaPurse: balance {"label":{"assay":{},"allegedName":"moola"},"extent":0}',
+  'bobSimoleanPurse;: balance {"label":{"assay":{},"allegedName":"simoleans"},"extent":17}',
+  'Order update: b:[], s:[]',
+  'The offer has been accepted. Once the contract has been completed, please check your payout',
+  'bobMoolaPurse: balance {"label":{"assay":{},"allegedName":"moola"},"extent":3}',
+  'bobSimoleanPurse;: balance {"label":{"assay":{},"allegedName":"simoleans"},"extent":15}',
+  'Order update: b:simoleans:2 for moola:8, s:[]',
+  'The offer has been accepted. Once the contract has been completed, please check your payout',
+  'bobMoolaPurse: balance {"label":{"assay":{},"allegedName":"moola"},"extent":3}',
+  'bobSimoleanPurse;: balance {"label":{"assay":{},"allegedName":"simoleans"},"extent":2}',
+  'Order update: b:simoleans:2 for moola:8,simoleans:13 for moola:20, s:[]',
+  'The offer has been accepted. Once the contract has been completed, please check your payout',
+  'bobMoolaPurse: balance {"label":{"assay":{},"allegedName":"moola"},"extent":3}',
+  'bobSimoleanPurse;: balance {"label":{"assay":{},"allegedName":"simoleans"},"extent":0}',
+  'aliceMoolaPurse: balance {"label":{"assay":{},"allegedName":"moola"},"extent":0}',
+  'aliceSimoleanPurse;: balance {"label":{"assay":{},"allegedName":"simoleans"},"extent":7}',
+];
+
+test('zoe - simpleExchange - state Update - with SES', async t => {
+  try {
+    const startingExtents = [
+      [3, 0, 0],
+      [0, 24, 0],
+    ];
+    const dump = await main(true, 'zoe', [
+      'simpleExchangeUpdates',
+      startingExtents,
+    ]);
+    t.deepEquals(dump.log, expectedSimpleExchangeUpdateLog);
+  } catch (e) {
+    t.isNot(e, e, 'unexpected exception');
+  } finally {
+    t.end();
+  }
+});
+
+test('zoe - simpleExchange - state Update - no SES', async t => {
+  try {
+    const startingExtents = [
+      [3, 0, 0],
+      [0, 24, 0],
+    ];
+    const dump = await main(false, 'zoe', [
+      'simpleExchangeUpdates',
+      startingExtents,
+    ]);
+    t.deepEquals(dump.log, expectedSimpleExchangeUpdateLog);
+  } catch (e) {
```

Comment:
Wrapping test inside try catch blocks is not necessary. Instead of `t.end()`, use `t.plan(1)` at the top and you can remove the `try-catch-finally`, the `t.isNot()` check on error, and the `t.end()`. The test framework does that for you.

Reply:
Thanks. I didn't know about .plan().